src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,418 @@
+        }
+
+        [Fact]
+        public void TestBadNonTrailing2()
+        {
+            var source = @"
+class C
+{
+    static void M(int a = 1, int b = 2, int c = 3)
+    {
+    }
+    static void M(long c = 1, long b = 2)
+    {
+        System.Console.Write($""Second {c} {b}. "");
+    }
+    static void Main()
+    {
+        int valueB = 2;
+        int valueC = 3;
+        M(c: valueC, valueB);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "Second 3 2.", parseOptions: TestOptions.Regular7_2);
+            verifier.VerifyDiagnostics();
+
+            var tree = verifier.Compilation.SyntaxTrees.First();
+            var model = verifier.Compilation.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(c: valueC, valueB)", invocation.ToString());
+            Assert.Equal("void C.M([System.Int64 c = 1], [System.Int64 b = 2])",
+                model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void TestBadNonTrailing3()
+        {
+            var source = @"
+class C
+{
+    static void M(long a = 1, long b = 2, long c = 3)
+    {
+    }
+    static void M(int c = 1, int b = 2)
+    {
+        System.Console.Write($""Second {c} {b}."");
+    }
+    static void Main()
+    {
+        int valueB = 2;
+        int valueC = 3;
+        M(c: valueC, valueB);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "Second 3 2.", parseOptions: TestOptions.Regular7_2);
+            verifier.VerifyDiagnostics();
+
+            var tree = verifier.Compilation.SyntaxTrees.First();
+            var model = verifier.Compilation.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(c: valueC, valueB)", invocation.ToString());
+            Assert.Equal("void C.M([System.Int32 c = 1], [System.Int32 b = 2])",
+                model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void TestParams()
+        {
+            var source = @"
+class C
+{
+    static void M(int a, int b, params int[] c)
+    {
+    }
+    static void Main()
+    {
+        M(b: 2, 3, 4);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (9,9): error CS7036: There is no argument given that corresponds to the required formal parameter 'a' of 'C.M(int, int, params int[])'
+                //         M(b: 2, 3, 4);
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("a", "C.M(int, int, params int[])").WithLocation(9, 9)
```

Comment:
>Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("a", "C.M(int, int, params int[])").WithLocation(9, 9) [](start = 16, length = 123)

I would expect to see ERR_BadNonTrailingNamedArgument error, instead of this one, or in addition to it.